### ISSUE TITLE
help needed about running multiple brew install command concurrently

### ISSUE THREAD

**Issue by eggio** [Homebrew/homebrew-core#3431]
I am interested in knowing a possibly better solution to a closed github issue named "Another active Homebrew process is already using xz? #47593". 

For installing some packages that require some particular commands, it will generate similar error messages to the above quoted one. For instance, xz command; I have been encountering this issue when I run the following command at the same time:
```
brew install tmux
brew install irssi
```

Without digging into the mechanism of homebrew, I am just curious about whether some reasonable amount of effort is worthwhile to obtain a real solution to this, and which part of the code takes care of "concurrency" and why it is not possible to let multiple homebrews use xz at one time at the current state?

The bottom line is I am interested in running multiple brew install to maximize the resource usage or shorten the overall installation time disregard the kinds of invoking commands or dependencies.

**Comment by MikeMcQuaid** [comment]
This could maybe be finer grained than it is currently but it's not something we're actively working on so you probably want to just try and see if you can improve it yourself.

Re: concurrency: we already use concurrency when building so you are unlike to make the bulk of the installation time any faster. You could also consider using `brew fetch` manually before you `brew install`.
